### PR TITLE
feat: do create drop table

### PR DESCRIPTION
**Pull request by ShiKaiWi**
# Which issue does this PR close?

Part of #235 

# Rationale for this change
 Do real job of creating/dropping table by meta client.
NOTE: This pr is blocked by https://github.com/CeresDB/ceresdbproto/pull/27.

# What changes are included in this PR?
- Add encoding for schema;
- Send creating/dropping table request to ceresmeta;

# Are there any user-facing changes?
None.

# How does this change test
- Add a new UT for schema encoding.
- Creating/Dropping table by ceresmeta will be tested by manual.

### CODE REVIEW

File: common_types/src/schema.rs [hunk 1]
```diff
@@ -151,18 +153,44 @@ pub enum Error {
 
     #[snafu(display("Arrow schema meta key not found.\nerr:\n{}", source))]
     ColumnSchemaDeserializeFailed { source: crate::column_schema::Error },
+
+    #[snafu(display("Failed to encode schema by protobuf, err:{}", source))]
+    EncodePbSchema {
```
Comment (Rachelint) [inline common_types/src/schema.rs:158]: EncodeSchemaToPb?

File: meta_client/src/meta_impl.rs [hunk 2]
```diff
@@ -132,10 +132,10 @@ impl MetaClient for MetaClientImpl {
         info!("Meta client finish creating table, resp:{:?}", pb_resp);
```
Comment (Rachelint) [inline meta_client/src/meta_impl.rs:132]: finish --> finished

File: meta_client/src/types.rs [hunk 3]
```diff
@@ -298,20 +305,29 @@ impl From<CreateTableRequest> for meta_service::CreateTableRequest {
...
-            shard_id: pb_resp.shard_id,
-            schema_id: pb_resp.schema_id,
-            id: pb_resp.id,
-        }
+impl TryFrom<meta_service::CreateTableResponse> for CreateTableResponse {
+    type Error = Error;
+
+    fn try_from(pb_resp: meta_service::CreateTableResponse) -> Result<Self> {
+        let pb_table_info = pb_resp.created_table.context(MissingTableInfo {
+            msg: "created table is not found in the create table response",
```
Comment (Rachelint) [inline meta_client/src/types.rs:321]: table requested to create is not found in the response ?

File: common_types/src/schema.rs [hunk 4]
```diff
@@ -151,18 +153,44 @@ pub enum Error {
...
+    DecodeSchemaFromPb {
+        buf: Vec<u8>,
+        source: protobuf::error::ProtobufError,
+    },
 }
 
 pub type CatalogName = String;
 pub type SchemaId = u32;
 pub type SchemaName = String;
+pub type Result<T> = std::result::Result<T, Error>;
```
Comment (Rachelint) [inline common_types/src/schema.rs:186]: Use `define_result!(Error)` to do this?

**Comment by ShiKaiWi** [comment]
> Sorry for the late reply, there are some troubling problems of my work yesterday. As I see, the complete process is like this:
> 
> ```
>                    +-----------+                  +-----------+              
>             |------>   node    |                  |   node    |-------------+
>             |      +-----------+                  +-----------+             |
>             |            |                              ^                   |
>             |    1.send  |                              |2.meta chooses     |
>    4.meta   |    create  |                              |a proper node      |
>    responds |    request |                              |and send it requst |
>    to the   |    to meta |                              |, make it do the   |
>    invoking |            |                              |real create work   |
>    node     |            |                              |                   |
>             |            |                              |                   |
>             |      +-----v-----+                        |                   |
>             +------|   meta    |------------------------+                   |
>                    +-----------+                                            |
>                           ^                                                 |
>                           |                    3.working node finished its  |
>                           |                    work and responds to meta    |
>                           +-------------------------------------------------|
> ```
> 
> So, this pr implemented the logic about 1 and 4?

Sorry for that I didn't give a full description about this PR, and your conclusion is exactly correct and concise.

File: meta_client/src/meta_impl.rs (hunk 2 above)
Comment (ShiKaiWi) [reply meta_client/src/meta_impl.rs:132]: Using `finished` here seems wrong in grammar.

File: meta_client/src/types.rs (hunk 3 above)
Comment (ShiKaiWi) [reply meta_client/src/types.rs:321]: Yes. Actually, this won't happen in the real implementation, but the protobuf message generated by prost make any sub message is optional which leading to such check here.

File: common_types/src/schema.rs (hunk 4 above)
Comment (ShiKaiWi) [reply common_types/src/schema.rs:186]: `define_result!` is defined in `common_util` crate, which depends on `common_types`, so here we can't use this macro.

File: meta_client/src/meta_impl.rs (hunk 2 above)
Comment (Rachelint) [reply meta_client/src/meta_impl.rs:132]: finished to create table?
I think such task has been done, past tense may be better.

**Review by Rachelint — APPROVED** [review]
LGTM

**Pull request merged**